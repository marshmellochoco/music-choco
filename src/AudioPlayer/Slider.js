export default function Slider({onChange, transform}){
    return(
        <div className="slider" >
            <div style={{width: '100%', height: '100%', backgroundColor: 'green', transform: `translateX(${transform}%)`}}/>
            <input type='range' style={{width: '100%', height: '100%'}} onChange={(e) => onChange(e)}/>
        </div>
    )
}