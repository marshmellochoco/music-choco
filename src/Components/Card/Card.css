:root {
    --primary-color: #404040;
    --secondary-color: #707070;
    --thirtiary-color: #a0a0a0;
    --contrast-color: #d0d0d0;
}

.cardItem{
    width: 200px;
    height: 300px;
    padding: 1% 1% 0.5% 1%;
    display: grid;
    grid-template-rows: 3fr 1fr 0.5fr;
    border-radius: 2%;
    background-color: var(--secondary-color);
    cursor: pointer;
    color: var(--contrast-color);
}

.cardItem:hover{
    animation: cardItemHover 0.4s forwards;
}

@keyframes cardItemHover {
    0%{
        opacity: 1;
    }
    100%{
        opacity: 0.8;
    }
}

.cardItem img{
    width: 200px;
}

.cardItem h1{
    font-size: 1.5em;
    margin: 0;
}

.cardItem p{
    overflow: hidden;
    font-size: 1.2em;
    margin: 0;
}