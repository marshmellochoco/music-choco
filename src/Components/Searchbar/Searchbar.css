.search{
    height: 2em;
    padding: 0 10px;
    display: grid;
    grid-template-columns: 24px 6fr;
    align-self: center;
    background-color: var(--contrast-color);
    border-radius: 2em;
}

.search .icon{
    align-self: center;
    margin: 0;
    fill: var(--primary-color);
    width: 24px;
    height: 24px;
}

.search input[type=text]{
    display: flex;
    height: 1em;
    font-size: 1em;
    align-self: center;
    border: none;
    background-color: var(--contrast-color);
    outline: none;
}