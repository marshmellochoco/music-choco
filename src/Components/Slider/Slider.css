.slider{
    width: 96%;
    margin: auto 2%;
    background-color: grey;
    height: 0.3em;
    border-radius: 2px;
    overflow: hidden;
}

.slider-front{
    width: 100%;
    height: 100%; 
    background-color: green;
}

.slider-back{
    cursor: pointer;
    width: 100%;
    height: 100%;
    opacity: 0;
    margin: auto 0;
    float: right;
    transform: translateY(-100%);
}