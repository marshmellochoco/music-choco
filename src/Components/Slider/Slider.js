// component import
import "./Slider.css";

export const Slider = ({ onChange, transform }) => {
    return (
        <div className="slider">
            <div
                className="slider-front"
                style={{ transform: `translateX(${transform}%)` }}
            />
            <input
                className="slider-back"
                type="range"
                onChange={(e) => onChange(e)}
            />
        </div>
    );
};
