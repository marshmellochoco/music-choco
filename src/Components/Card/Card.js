import { Link } from "react-router-dom";
import "./Card.css";

export const Card = ({ id, name, artist }) => {
    return (
        <Link className="cardItem" to={`/albums/${id}`}>
            <img
                src={"http://localhost:4000/album/" + id + "/ico"}
                alt="Album Icon"
            ></img>
            <h1>{name}</h1>
            <p>{artist}</p>
        </Link>
    );
};
