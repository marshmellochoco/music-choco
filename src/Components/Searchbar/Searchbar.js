// component import
import { mdiMagnify } from "@mdi/js";
import Icon from "@mdi/react";
import "./Searchbar.css";

export const Searchbar = ({ handleSearch }) => {
    return (
        <div className="search">
            <Icon path={mdiMagnify} className="icon" />
            <input
                type="text"
                onChange={(e) => handleSearch(e.target.value)}
            ></input>
        </div>
    );
};
