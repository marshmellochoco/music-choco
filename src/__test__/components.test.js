import { configure, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";
import { Route, BrowserRouter as Router } from "react-router-dom";

import { Card } from "../Components/Card/Card";
import { Searchbar } from "../Components/Searchbar/Searchbar";
import { Slider } from "../Components/Slider/Slider";

configure({ adapter: new Adapter() });

describe("components test", () => {
    it("should render card correctly", () => {
        const location = { pathname: "/music-choco" };
        const wrapper = mount(
            <Router>
                <Route location={location}>
                    <Card id={"60883d8e1a02ff3eac90191f"} />
                </Route>
            </Router>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should render searchbar correctly", () => {
        const wrapper = mount(<Searchbar />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should be able to search", () => {
        const mockSearch = jest.fn();
        const wrapper = mount(<Searchbar handleSearch={mockSearch} />);
        wrapper
            .find("input")
            .simulate("change", { target: { value: "search" } });
        expect(mockSearch.mock.calls.length).toBe(1);
        expect(mockSearch).toBeCalledWith("search");
    });

    it("should render slider correctly", () => {
        const wrapper = mount(<Slider transform={50} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should be able to slide", () => {
        const mockChange = jest.fn();
        const wrapper = mount(<Slider onChange={mockChange} transform={50} />);
        wrapper.find("input").simulate("change", { target: { value: 20 } });
        expect(mockChange.mock.calls.length).toBe(1);
    });
});
