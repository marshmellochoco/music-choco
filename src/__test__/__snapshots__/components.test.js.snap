// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components test should render card correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Route
        location={
          Object {
            "pathname": "/music-choco",
          }
        }
      >
        <Card
          id="60883d8e1a02ff3eac90191f"
        >
          <Link
            className="cardItem"
            to="/albums/60883d8e1a02ff3eac90191f"
          >
            <LinkAnchor
              className="cardItem"
              href="/albums/60883d8e1a02ff3eac90191f"
              navigate={[Function]}
            >
              <a
                className="cardItem"
                href="/albums/60883d8e1a02ff3eac90191f"
                onClick={[Function]}
              >
                <img
                  alt="Album Icon"
                  src="http://118.100.207.12:4000/api/album/ico/60883d8e1a02ff3eac90191f"
                />
                <h1 />
                <div />
                <p />
                <div
                  className="playAlbum"
                  onClick={[Function]}
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clipRule="evenodd"
                      d="M10 18a8 8 0 100-16 8 8 0 000 16zM9.555 7.168A1 1 0 008 8v4a1 1 0 001.555.832l3-2a1 1 0 000-1.664l-3-2z"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </a>
            </LinkAnchor>
          </Link>
        </Card>
      </Route>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`components test should render searchbar correctly 1`] = `
<Searchbar>
  <div
    className="search"
  >
    <svg
      className="icon"
      viewBox="0 0 24 24"
    >
      <path
        d="M9.5,3A6.5,6.5 0 0,1 16,9.5C16,11.11 15.41,12.59 14.44,13.73L14.71,14H15.5L20.5,19L19,20.5L14,15.5V14.71L13.73,14.44C12.59,15.41 11.11,16 9.5,16A6.5,6.5 0 0,1 3,9.5A6.5,6.5 0 0,1 9.5,3M9.5,5C7,5 5,7 5,9.5C5,12 7,14 9.5,14C12,14 14,12 14,9.5C14,7 12,5 9.5,5Z"
      />
    </svg>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>
</Searchbar>
`;

exports[`components test should render slider correctly 1`] = `
<Slider
  transform={50}
>
  <div
    className="slider"
  >
    <div
      className="slider-front"
      style={
        Object {
          "transform": "translateX(50%)",
        }
      }
    />
    <input
      className="slider-back"
      onChange={[Function]}
      type="range"
    />
  </div>
</Slider>
`;
